.github/workflows/test.yml

```diff
@@ -22,5 +22,15 @@ jobs:
       ###########################################
       - name: build-docker-test
         run: |
-          docker build -t cpp . -f etc/docker/tests/Dockerfile.debian.bullseye
-          docker run -e BUILD_SHARED=${{ matrix.BUILD_SHARED }} --rm -i -w /tmp cpp /bin/bash
+          docker build -t cpp . -f etc/docker/base-images/Dockerfile.debian.bullseye
```

Comment:
is this creating a base image if you don't already have one? otherwise what is this `docker build `doing compared to the one below it/why do we need two?

Reply:
yeah this is creating the base image which the step below then uses to build the SDK. The idea is to have a modular, many-to-one type scenario where we can have a bunch of base images (the ones in `base-images`) which, if built, can then be used as input to a single "build the sdk on top of a base" image. from what I found researching this requires having stuff in separate Dockerfiles which are in turn built with separate invocations of `docker build`